app/policies/idv/flow_policy.rb

```diff
@@ -0,0 +1,158 @@
+module Idv
+  class FlowPolicy
+    include Rails.application.routes.url_helpers
+
+    NEXT_STEPS = Hash.new([])
+    NEXT_STEPS.merge!(
+      {
+        root: [:welcome],
+        welcome: [:agreement],
+        agreement: [:hybrid_handoff, :document_capture],
+        hybrid_handoff: [:link_sent, :document_capture],
+        link_sent: [:ssn],
+        document_capture: [:ssn], # in person?
+        ssn: [:verify_info],
+        verify_info: [:phone],
+        phone: [:phone_enter_otp],
+        phone_enter_otp: [:review],
+        review: [:personal_key],
+        personal_key: [:success],
+      },
+    )
+
+    attr_reader :idv_session, :user
+
+    def initialize(idv_session:, user:)
+      @idv_session = idv_session
+      @user = user
+    end
+
+    def step_allowed?(step:)
+      send(step)
+    end
+
+    def path_allowed?(path:)
+      step = path_to_step[path]
+      send(step)
+    end
+
+    def latest_step(current_step: :root)
+      return nil if NEXT_STEPS[current_step].empty?
+      return current_step if NEXT_STEPS[current_step] == [:success]
+
+      (NEXT_STEPS[current_step]).each do |step|
+        if step_allowed?(step: step)
+          return latest_step(current_step: step)
+        end
+      end
+      current_step
+    end
+
+    def path_for_latest_step
+      path_map[latest_step]
+    end
+
+    # This can be blank because we are using paths, not urls,
+    # so international urls are supported.
+    def url_options
+      {}
+    end
+
+    private
+
+    def path_to_step
+      @path_to_step ||= {
+        idv_welcome_path => :welcome,
+        idv_agreement_path => :agreement,
+        idv_hybrid_handoff_path => :hybrid_handoff,
+        idv_link_sent_path => :link_sent,
+        idv_document_capture_path => :document_capture,
+        idv_ssn_path => :ssn,
+        idv_verify_info_path => :verify_info,
+        idv_phone_path => :phone,
+        idv_otp_verification_path => :phone_enter_otp,
+        idv_review_path => :review,
+        idv_personal_key_path => :personal_key,
+      }.freeze
+    end
+
+    def path_map
+      @path_map ||= {
+        welcome: idv_welcome_path,
+        agreement: idv_agreement_path,
+        hybrid_handoff: idv_hybrid_handoff_path,
+        link_sent: idv_link_sent_path,
+        document_capture: idv_document_capture_path,
+        ssn: idv_ssn_path,
+        verify_info: idv_verify_info_path,
+        phone: idv_phone_path,
+        phone_enter_otp: idv_otp_verification_path,
+        review: idv_review_path,
+        personal_key: idv_personal_key_path,
+      }.freeze
+    end
```

Comment:
It looks like these hashes are the opposite of each other?

We could likely define them once and use [`Hash#invert`](https://docs.ruby-lang.org/en/3.2/Hash.html#method-i-invert) to create the other?